File: Excel/1000-1099/1047/Challenge 1047.R
library(tidyverse)
library(readxl)

path <- "1000-1099/1047/1047 Cryptographic Challenge.xlsx"
input <- read_excel(path, range = "A1:A10")
test <- read_excel(path, range = "B1:B10")

f <- \(s) {
  str_split_1(s, "") |>
    enframe(name = "i", value = "x") |>
    arrange(desc(i %% 2), if_else(i %% 2 == 1, -i, i)) |>
    mutate(x = LETTERS[(match(x, LETTERS) + row_number() - 1) %% 26 + 1]) |>
    pull(x) |>
    str_c(collapse = "")
}

result <- input %>%
  mutate(`Answer Expected` = map_chr(Message, f))

all.equal(result$`Answer Expected`, test$`Answer Expected`)
# TRUE

File: Excel/1000-1099/1047/Challenge 1047.py
import pandas as pd

path = "1000-1099/1047/1047 Cryptographic Challenge.xlsx"
input = pd.read_excel(path, usecols="A", nrows=9)
test = pd.read_excel(path, usecols="B", nrows=9)


def f(s):
    x = s[::2][::-1] + s[1::2]
    return "".join(chr((ord(c) - 65 + i) % 26 + 65) for i, c in enumerate(x, 1))


input["Answer Expected"] = input["Message"].map(f)

print(input["Answer Expected"].equals(test["Answer Expected"]))
# True
